src/StdLib.jl

```diff
@@ -126,27 +126,41 @@ Base.size(v::StdValArray) = (Int(cppsize(v)),)
 Base.getindex(v::StdValArray, i::Int) = cxxgetindex(v,i)[]
 Base.setindex!(v::StdValArray{T}, val, i::Int) where {T} = cxxsetindex!(v, convert(T,val), i)
 
+
+# Deque
 function StdDeque() where {T}
   return StdDeque{T}()
 end
-
 Base.IndexStyle(::Type{<:StdDeque}) = IndexLinear()
-Base.size(d::StdDeque) = (Int(cppsize(d)),)
-Base.resize!(d::StdDeque, n::Integer) = resize(d, n)
-Base.getindex(d::StdDeque, i::Int) = cxxgetindex(d, i)[]
-Base.setindex!(d::StdDeque{T}, val, i::Int) where {T} = cxxsetindex(d, convert(T, val), i)
-#TODO: edit the cxx part to enable push to get more than two arguments
-Base.push!(d::StdDeque, x) = push_back(d, x)
-Base.pushfirst!(d::StdDeque, x) = push_front(d, x)
-Base.pop!(d::StdDeque) = pop_back(d)
-Base.popfirst!(d::StdDeque) = pop_front(d)
-Base.isempty(d::StdDeque) = isEmpty(d)
-Base.empty!(d::StdDeque) = clear(d)
-
-# Iteration utilities
-Base.:(==)(a::StdIterator, b::StdIterator) = iterator_is_equal(a, b)
-_deque_iteration_tuple(d::StdDeque, state::StdIterator) = (state == iteratorend(d)) ? nothing : (iterator_value(state), state)
-Base.iterate(d::StdDeque) = _deque_iteration_tuple(d, iteratorbegin(d))
-Base.iterate(d::StdDeque, state::StdIterator) = (state != iteratorend(d)) ? _deque_iteration_tuple(d, iterator_next(state)) : nothing
-#TODO:remove the iterator_value method from the cxx part, since it is not needed
-end # module
+Base.size(v::StdDeque) = (Int(cppsize(v)),)
```

Comment:
Thank you for your patch!

However, I would recommend to not mix actual changes with changes that rename arguments or reorder code, as it makes it needlessly difficult to review your changes.

Here, you changed all `d::StdDeque` to `v::StdDeque` which has no obvious benefit in itself. Perhaps you prefer it, but then you should put it into a separate commit, or even separate PR. The same applies to reordering the methods. Together the result is that it is very difficult to see if you changed anything in the implementation of these methods, and what exactly.

I would really prefer if this part of the change was undone, but that's just me, @barche may disagree :-)

Reply:
Thanks for the advice, I will stick to it in the future.
I renamed the parameters back to the `d` and the ordering was my fault as it was not aligned with the ordering of the C++ part of the component.